scripts/g.extension.all/g.extension.all.py

```diff
@@ -77,6 +88,124 @@ def get_extensions():
     return ret
 
 
+def urlopen(url, *args, **kwargs):
+    """Wrapper around urlopen. Same function as 'urlopen', but with the
+    ability to define headers.
+    """
+    request = urlrequest.Request(url, headers=HEADERS)
+    return urlrequest.urlopen(request, *args, **kwargs)
+
+
+def download_modules_xml_file(url, response_format, *args, **kwargs):
+    """Generates JSON file containing the download URLs of the official
+    Addons
+
+    :param str url: url address
+    :param str response_format: content type
+
+    :return response: urllib.request.urlopen response object or None
+    """
+    try:
+        response = urlopen(url, *args, **kwargs)
+
+        if not response.code == 200:
+            index = HTTP_STATUS_CODES.index(response.code)
+            desc = HTTP_STATUS_CODES[index].description
+            gscript.fatal(
+                _(
+                    "Download file from <{url}>, "
+                    "return status code {code}, "
+                    "{desc}".format(
+                        url=url,
+                        code=response.code,
+                        desc=desc,
+                    ),
+                ),
+            )
+        if response_format not in response.getheader("Content-Type"):
+            gscript.fatal(
+                _(
+                    "Wrong downloaded file format. "
+                    "Check url <{url}>. Allowed file format is "
+                    "{response_format}.".format(
+                        url=url,
+                        response_format=response_format,
+                    ),
+                ),
+            )
+        return response
+
+    except HTTPError as err:
+        if err.code == 404:
+            gscript.fatal(
+                _(
+                    "The download of the modules.xml file "
+                    "from the osgeo server wasn't successful. "
```

Comment:
```suggestion
                    "from the server grass.osgeo.org was not successful. "
```

Reply:
I find "osgeo" somehow confusing in this context, and the URL comes anyway in the next line...